### PR TITLE
Add File.touch and FileUtils.touch methods

### PR DESCRIPTION
**Pull request by Sija**
As in the title. Low hanging fruit since `File.utime` is now in stdlib.

### CODE REVIEW

File: src/file.cr [hunk 1]
```diff
@@ -595,6 +595,18 @@ class File < IO::FileDescriptor
...
 
+  # Attempts to set the access and modification times of the file named
+  # in the *filename* parameter to the value given in *time*.
+  #
+  # If the file does not exist, it will be created.
+  def self.touch(filename : String, time : Time? = nil)
+    unless exists?(filename)
+      File.open(filename, "a") { }
+    end
+    time ||= Time.now
```
Comment (RX14) [inline src/file.cr]: Why not put ` = Time.now` as the default argument?

File: src/file_utils.cr [hunk 2]
```diff
@@ -67,6 +67,34 @@ module FileUtils
...
+
+  # Attempts to set the access and modification times of each file given
+  # in the *paths* parameter to the value given in *time*.
+  #
+  # If the file does not exist, it will be created.
+  #
+  # ```
+  # FileUtils.touch(["foo", "bar"])
+  # ```
+  def touch(paths : Enumerable(String), time : Time? = nil)
```
Comment (RX14) [inline src/file_utils.cr]: Maybe a `*paths` version?

File: src/file.cr (hunk 1 above)
Comment (Sija) [inline src/file.cr]: Since this way you might still pass a `nil` as an argument.

File: src/file_utils.cr (hunk 2 above)
Comment (Sija) [inline src/file_utils.cr]: Sounds good, though I'm just following convention from `FileUtils` here.

File: src/file.cr (hunk 1 above)
Comment (ysbaddaden) [inline src/file.cr]: Why would someone pass `nil`?

Comment (Sija) [inline src/file.cr]: I have no strong opinion about it, might as well change it to `= Time.now`.

**Comment by Sija** [comment]
Ping? Anything actionable here?

**Comment by luislavena** [comment]
@Sija I think the removal of the support for `nil` should be enough. Either you provide a `Time` reference or  let the default `Time.now` be used.

**Comment by Sija** [comment]
@luislavena 'fair nuff, done!

**Comment by Sija** [comment]
Is this PR is ready to 🚢 ?

**Comment by bcardiff** [comment]
So, should we redo some changes applied on #3986 since there is a built in touch?

**Comment by Sija** [comment]
@bcardiff its usage is part of a broader `Process.exec` method testing and thus using here `File.touch` would make the test miss its point.

**Comment by bcardiff** [comment]
@Sija yes, you are right.

**Comment by rdp** [comment]
File feels like the right place for this, thank you for fixing another ruby oddity :)

**Pull request merged**